assignment3/word_search.py

```diff
@@ -0,0 +1,93 @@
+#! /usr/bin/python3.5
+
+import dictionary
+ALL_WORDS = set()
+WORD_DICT = None
+CHAR_2D_ARRAY = None
+
+def findAllWords(row, col, char2dArray, wordDict):
+    """Find all valid words that can be formed in the char2dArray. 
+    Note that CHAR_2D_ARRAY, WORD_DICT and ALL_WORDS are global variables. 
```

Comment:
I don't suppose there is any reason to prefer global variables over wrapping all this into a `WordSearcher` class?
Global singleton variables are worth avoiding.

They are more awkward to deal with (with the `global` keyword below). And they only allow one instance of word searching to occur at the same time. Not a problem for this assignment of course.